packages/sdk-bridge/src/backend.ts

```diff
@@ -0,0 +1,281 @@
+        let m = await this.getMessage(messageHash);
+        if (!m?.receive_tx) m = await this.getMessage(messageHash, true);
+        return m?.receive_tx;
+    }
+
+    /**
+     * Fetches internal message from backend
+     *
+     * @returns Internal message representation (if any)
+     */
+     async fetchMessages(f: Partial<MessageFilter>, limit?: number): Promise<GoldSkyBridgeMessage[]|undefined> {
+
+        const query = gql`
+          query Query($committedRoot: String, $messageHash: String, $transactionHash: String, $limit: Int) {
+
+                  bridge_events(where: {_or: [{dispatch_tx: {_eq: $transactionHash}}, {message_hash: {_eq: $messageHash}}, {old_root: {_eq: $committedRoot}}]}, limit: $limit) {
+                    committed_root
+                    destination_and_nonce
+                    destination_domain_id
+                    destination_domain_name
+                    dispatch_block
+                    dispatch_tx
+                    dispatched_at
+                    id
+                    leaf_index
+                    message
+                    message__action__amount
+                    message__action__details_hash
+                    message__action__to
+                    message__action__type
+                    message__token__domain
+                    message__token__id
+                    message_body
+                    message_hash
+                    message_type
+                    new_root
+                    nonce
+                    old_root
+                    origin_domain_id
+                    origin_domain_name
+                    process_block
+                    process_tx
+                    processed_at
+                    recipient_address
+                    relay_block
+                    relay_chain_id
+                    relay_tx
+                    relayed_at
+                    sender_address
+                    signature
+                    update_block
+                    update_chain_id
+                    update_tx
+                    updated_at
+
+                    send_tx
+                    sent_at
+                    send_block
+                    original_sender
+                    
+                    receive_tx
+                    origin_and_nonce
+                    receive_block
+                    received_at
+                  }
+          }
+        `;
+
+        const headers = {
+          "content-type": "application/json",
+          "x-hasura-admin-secret": this._secret,
+        };
+
+        const filter = {
+            ...GoldSkyBackend.fillFilter(f),
+            limit: limit || null
+        };
+
+      const response = await request(this.uri, query, filter, headers);
+
+      const events = response.bridge_events;
+
+      if (events.length <= 0) return undefined;
+
+        return events.map((event: any) => ({
+            committed_root: event.committed_root,
```

Comment:
Are there specific fields we're renaming here or that we don't want from the event? Just curious

Reply:
Addressed in the new commit